NodesPair[] map
string source
---
NodeInfo[] result
---
NodeInfo current_node